org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/desktop/DashboardController.java

```diff
@@ -961,6 +987,38 @@ public boolean render(Component content, MDashboardContent dashboardContent, Das
     	return !empty;
 	}
 	
+	/**
+	 * Add Drill Across Event Listener to Border Layout
+	 * @param processID
+	 */
+	private void addDrillAcrossEventListener(int processID) {
+		Component parent = dcParent;
+		while(parent.getParent() != null) {
+			if(parent instanceof Borderlayout) {
+				parent.addEventListener(DrillEvent.ON_DRILL_ACROSS, new EventListener<Event>() {
+    				
+    				public void onEvent(Event event) throws Exception {
+    					if (event instanceof DrillEvent) {
+    						Clients.clearBusy();
+    						DrillEvent de = (DrillEvent) event;
+    						if (de.getData() != null && de.getData() instanceof DrillData) {
+    							DrillData data = (DrillData) de.getData();
+    							if(data.getData() instanceof JSONArray) {
+    								JSONArray jsonData = (JSONArray) data.getData();
+    								if(jsonData.indexOf(String.valueOf(processID)) < 0)
+    									return;
+    							}
+    							AEnv.actionDrill(data, m_WindowNo, processID);
+    						}
+    					}
+    				}
+    			});
+				break;
+			}
+			parent = parent.getParent();
+		}
+	}
```

Comment:
this is a strange one, why must add to border layout instead of the dashboard gadget panel ifself ?

Reply:
I've checked in DrillCommand.java:47  the target component `comp = request.getComponent()` is BorderLayout

Reply:
should simplify to appDesktop.getComponent().addEventListener(...), i.e it must be the same as what's use in the generateReport method.